application/src/main/java/org/togetherjava/tjbot/config/JShellConfig.java

```diff
@@ -0,0 +1,44 @@
+package org.togetherjava.tjbot.config;
+
+import com.linkedin.urls.Url;
+
+import org.togetherjava.tjbot.features.utils.RateLimiter;
+
+import java.net.MalformedURLException;
+
+/**
+ * JShell config.
+ * 
+ * @param baseUrl the base url of the JShell REST API
+ * @param rateLimitWindowSeconds the number of seconds of the {@link RateLimiter rate limiter} for
+ *        jshell commands and code actions
+ * @param rateLimitRequestsInWindow the number of requests of the {@link RateLimiter rate limiter}
+ *        for jshell commands and code actions
+ */
+public record JShellConfig(String baseUrl, int rateLimitWindowSeconds,
+        int rateLimitRequestsInWindow) {
+    /**
+     * Creates a JShell config.
+     * 
+     * @param baseUrl the base url of the JShell REST API, must be valid
+     * @param rateLimitWindowSeconds the number of seconds of the {@link RateLimiter rate limiter}
+     *        for jshell commands and code actions, must be higher than 0
+     * @param rateLimitRequestsInWindow the number of requests of the {@link RateLimiter rate
+     *        limiter} for jshell commands and code actions, must be higher than 0
```

Comment:
I don't believe it's needed

Reply:
`throws` is usually only added if u want to explain sth about it (or if its a checked exception, since then its necessary to explain).
but `IllegalArgumentException` when the arguments are illegal is just classic validation stuff - id consider it idiomatic enough to go without explanation